src/modules/reserve-overview/ReserveActions.tsx

```diff
@@ -74,35 +76,42 @@ export const ReserveActions = ({ reserve }: ReserveActionsProps) => {
     balance = walletBalances[API_ETH_MOCK_ADDRESS.toLowerCase()];
   }
 
-  const maxAmountToBorrow = getMaxAmountAvailableToBorrow(
-    reserve,
-    user,
-    InterestRate.Variable
-  ).toString();
+  let maxAmountToBorrow = getMaxAmountAvailableToBorrow(reserve, user, InterestRate.Variable);
+  const availableBorrowCap =
+    reserve.borrowCap === '0'
+      ? valueToBigNumber(ethers.constants.MaxUint256.toString())
+      : valueToBigNumber(reserve.borrowCap).minus(valueToBigNumber(reserve.totalDebt));
+  maxAmountToBorrow = BigNumber.max(BigNumber.min(maxAmountToBorrow, availableBorrowCap), 0);
```

Comment:
I think this calculation here for the borrow cap is redundant since it's also happening in `getMaxAmountAvailableToBorrow`